### ISSUE TITLE
Problem with logrotate config (PID discovery)

### ISSUE THREAD

**Issue by Ziars** [firehol/netdata#4020]
Greetings!

I've discovered a problem with current logrotate config behaviour on a server with a bunch of LXD containers. Here's the problematic line: https://github.com/firehol/netdata/blob/6832ce444d2637173a6bac436805ca9e55b84fc1/system/netdata.logrotate.in#L10

When executed on host system, pidof grabs all the PID's of netdata-slaves running in LXC-containers. For example:

```
~# ps aux | grep '[n]etdata -P'
100000     330  0.4  0.0 119112  4552 ?        Sl   Jul26  27:23 /usr/sbin/netdata -P /var/run/netdata.pid
root       832  2.6  0.4 360888 132544 ?       Sl   Jun20 1530:03 /usr/sbin/netdata -P /var/run/netdata.pid
100000   11231  0.7  0.0 124032 17492 ?        Sl   Jul05 279:17 /usr/sbin/netdata -P /var/run/netdata.pid
100000   13089  0.0  0.0  97648  1164 ?        Sl   Mar30 173:25 /usr/sbin/netdata -P /var/run/netdata.pid
100000   14501  0.5  0.0 111836  4036 ?        Sl   Jul15 114:58 /usr/sbin/netdata -P /var/run/netdata.pid
100000   15821  0.5  0.0 124280  5420 ?        Sl   Jul05 221:28 /usr/sbin/netdata -P /var/run/netdata.pid
100000   23054  0.5  0.0 111432  4356 ?        Sl   Jul05 200:56 /usr/sbin/netdata -P /var/run/netdata.pid
100000   29453  0.0  0.0 121936  2356 ?        Sl   Jun20  41:44 /usr/sbin/netdata -P /var/run/netdata.pid
100000   29691  0.6  0.0 123596  4812 ?        Sl   Jul05 224:10 /usr/sbin/netdata -P /var/run/netdata.pid
100000   30283  0.5  0.0 119368  4344 ?        Sl   Jul04 188:30 /usr/sbin/netdata -P /var/run/netdata.pid
100000   32069  0.5  0.0 111732  5424 ?        Sl   Jul05 190:12 /usr/sbin/netdata -P /var/run/netdata.pid

~# pidof netdata
32069 30283 29691 29453 23054 15821 14501 13089 11231 832 330
```

So, a SIGHUP is being sent to a series of wrong PID's, in addition to the needed one. As a solution, one can consider adding a -c option to pidof (from man):

>        -c     Only return process ids that are running with the same root directory.  This option is ignored for non-root users, as they will be unable to check the cur‐
>               rent root directory of processes they do not own.

Here's the result:

```
~# ps aux | grep '[n]etdata -P'
100000     330  0.4  0.0 119112  4552 ?        Sl   Jul26  27:23 /usr/sbin/netdata -P /var/run/netdata.pid
root       832  2.6  0.4 360888 132544 ?       Sl   Jun20 1530:03 /usr/sbin/netdata -P /var/run/netdata.pid
100000   11231  0.7  0.0 124032 17492 ?        Sl   Jul05 279:17 /usr/sbin/netdata -P /var/run/netdata.pid
100000   13089  0.0  0.0  97648  1164 ?        Sl   Mar30 173:25 /usr/sbin/netdata -P /var/run/netdata.pid
100000   14501  0.5  0.0 111836  4036 ?        Sl   Jul15 114:58 /usr/sbin/netdata -P /var/run/netdata.pid
100000   15821  0.5  0.0 124280  5420 ?        Sl   Jul05 221:29 /usr/sbin/netdata -P /var/run/netdata.pid
100000   23054  0.5  0.0 111432  4356 ?        Sl   Jul05 200:56 /usr/sbin/netdata -P /var/run/netdata.pid
100000   29453  0.0  0.0 121936  2356 ?        Sl   Jun20  41:44 /usr/sbin/netdata -P /var/run/netdata.pid
100000   29691  0.6  0.0 123596  4812 ?        Sl   Jul05 224:10 /usr/sbin/netdata -P /var/run/netdata.pid
100000   30283  0.5  0.0 119368  4344 ?        Sl   Jul04 188:30 /usr/sbin/netdata -P /var/run/netdata.pid
100000   32069  0.5  0.0 111732  5424 ?        Sl   Jul05 190:12 /usr/sbin/netdata -P /var/run/netdata.pid

~# pidof -c netdata
832
```

Could you please fix this in future releases?

**Comment by paulfantom** [comment]
@n0guest please don't spam. Read [how to use github reactions.](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/)

**Comment by n0guest** [comment]
@ktsaou, would you prefer PR for this fix?

**Comment by ktsaou** [comment]
Yes please.

**Comment by n0guest** [comment]
Sent as #4103. Please, check it.

Thank you.

**Comment by ktsaou** [comment]
This issues has been fixed with #4157 
Thank you!
